### DISCUSSION TITLE
PSR12 - spaces and accessibility

### DISCUSSION THREAD

**Discussion by brianteeman** [joomla/joomla-cms#38408] [General]
@HLeithner recently did the hard working of converting Joomla codestyle to PSR12.

I'm a big fan of standardising and even prefer adopting and enforcing a consistent standard oveer my personal preference. So I didn't say anything about the change from tabs to spaces. And truth be told I didn't really see this as anything more than a cosmetic change and a longstanding joke debate.

However it has recently been brought to my attention that spaces are not as good as tabs for accessibility so I would like to suggest that this part of the psr12 standard is not adopted.

Please read this article which explains how defaulting to tabs instead of spaces provides an [accessible first environment](https://alexandersandberg.com/articles/default-to-tabs-instead-of-spaces-for-an-accessible-first-environment/)

**Comment by brianteeman** [comment]
@chmst your response is appreciated as JAT lead

**Reply by carcam** [reply]
Hi @brianteeman thanks you very much for starting this discussion. We did notice this change and we had some internal discussions about it. Actually we also have discussed the possibility to add an exception to our ruleset for this. We are checking also other aspects so that we can make a proper proposal to allow a more inclusive coding.

**Reply by brianteeman** [reply]
thanks - after 3 weeks with no response I assumed no one was listening

**Reply by carcam** [reply]
oh!! I'm not that active on GH and for sure I do not check discussions and probably Christiane didn't notice till today.

But yes, we want Joomla to be accessible for visitors, site developers, content creators and of course for programmers so we need to address this ;)

**Comment by chmst** [comment]
We discussed it in JAT already, and came to the same conclusion. My apologies that I forgot commenting here.

**Comment by brianteeman** [comment]
Has anything happened with this?

**Reply by crystalenka** [reply]
It was discussed again in PLT. 

It seems that without "actual research", the PHP-FIG is not open to revising the PSR-12 standard: 

https://github.com/php-fig/per-coding-style/issues/25#issuecomment-1244410338

I don't know where we could go from here beyond sourcing research to support a proposal there.

**Comment by brianteeman** [comment]
I updated the manual with reference material for developers regarding this
